doc/vlan/SONiC_VLAN_trunk_all_except_none_HLD.md

```diff
@@ -0,0 +1,701 @@
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#Revision)
+  * [About this Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+  * [1 Feature Overview](#1-feature-overview)
+    * [1.1 Requirements](#1.1-requirements)
+      * [1.1.1 Configuration and Management Requirements](#1.1.1-configuration-and-management-requirements)
+    * [1.2 Design Overview](#1.2-design-overview)
+       * [1.2.1 Basic Approach](#1.2.1-basic-approach)
+       * [1.2.2 Container](#1.2.2-container)
+  * [2 Functionality](#2-functionality)
+    * [2.1 Target Deployment Use Case](#2.1-target-deployment-use-case)
+    * [2.2 Functional Description](#2.2-functional-description)
+  * [3 Design](#3-design)
+    * [3.1 Overview](#3.1-overview)
+    * [3.2 DB Changes](#3.2-db-changes)
+      * [3.2.1 CONFIG DB](#3.2.1-config-db)
+      * [3.2.2 APP DB](3.2.2-app-db)
+      * [3.2.3 STATE DB](3.2.3-state-db)
+    * [3.3 User Interface](3.3-user-interface)
+      * [3.3.1 Data Models](3.3.1-data-models)
+      * [3.3.2 CLI](3.3.2-cli)
+        * [3.3.2.1 Configuration Commands](3.3.2.1-configuration-commands)
+        * [3.3.2.2 Show Commands](3.3.2.2-show-commands)
+          * [3.3.2.2.1 show running-config](3.3.2.3.1-show-running-config)
+        * [3.3.2.3 IS-CLI Compliance](3.3.2.5-iscli-compliance)
+      * [3.3.3 REST API Support](3.3.3-rest-api-support)
+  * [4 Error Handling](5-error-handling)
+  * [5 Unit Test](8-unit-test)
+
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 01/04/2021  |   Haemanthi Sree K R, Tejaswi Goel     | Initial draft          |
+| 0.2 | 01/11/2021  |   Haemanthi Sree K R, Tejaswi Goel     | v1.0                   |
+
+# About this Manual
+This document introduces trunk port allowed VLANs configuration via KLISH CLI using the options: all, except, none or *vlan-list*. And discusses the corresponding changes required in mgmt-framework. 
+# Scope
+Covers the backend mechanism required to support each command and the unit testcases.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| VLAN                     | Virtual Local Area Network          |
+
+# 1 Feature Overview
+
+Provide ability to configure allowed VLANs on trunk port using options "all", "except", "none" or *vlan-list*. 
+
+## 1.1 Requirements
+
+### 1.1.1 Configuration and Management Requirements
+    
+1. Support KLISH CLI "all" option to allow all VLAN IDs on trunk port. Port is added in all existing VLANs and any future VLAN that get created in the system, port is automatically added in this VLAN. 
```

Comment:
Assuming that switchport mode trunk command is executed, what are the vlans allowed by default